zhaquirks/lidl/ts011f_plug.py

```diff
@@ -0,0 +1,127 @@
+"""LIDL TS011F plug."""
+
+import asyncio
+
+from zigpy.profiles import zha
+from zigpy.zcl.clusters.general import (
+    Basic,
+    GreenPowerProxy,
+    Groups,
+    Identify,
+    OnOff,
+    Ota,
+    Scenes,
+    Time,
+)
+
+from zhaquirks.const import (
+    DEVICE_TYPE,
+    ENDPOINTS,
+    INPUT_CLUSTERS,
+    MODEL,
+    OUTPUT_CLUSTERS,
+    PROFILE_ID,
+)
+from zhaquirks.tuya.ts011f_plug import Plug_3AC_4USB
+
+
+async def tuya_magic_spell_wand(dev, tries=100) -> None:
+    """Initialize device so that all endpoints become available."""
+    import inspect
+
+    basic_cluster = dev.endpoints[1].in_clusters[0]
+
+    # The magic spell is needed only once.
+    # TODO: Improve by doing this only once (successfully).
+
+    # Magic spell - part 1
+    attr_to_read = [4, 0, 1, 5, 7, 0xFFFE]
+    if "tries" in inspect.getfullargspec(basic_cluster.read_attributes)[0]:
+        await basic_cluster.read_attributes(attr_to_read, tries=tries)
+    else:
+        await basic_cluster.read_attributes(attr_to_read)
+
+    # Magic spell - part 2 (skipped - does not seem to be needed)
+    # attr_to_write={0xffde:13}
+    # basic_cluster.write_attributes(attr_to_write)
+
+    # Leave with rejoin - may need to be adjuste to work everywhere
+    # or require a minimum zigpy version
+    # This should have the device leave and rejoin immediately triggering
+    # the discovery of the endpoints that appear after the magic trick
+    await dev.zdo.request(0x0034, dev.ieee, 0x01, tries)
+
+    app = dev.application
+    # Delete the device from the database
+    app.listener_event("device_removed", dev)
+
+    # Delete the device from zigpy
+    app.devices.pop(dev.ieee, None)
+
+
+def tuya_magic_spell(dev, tries=100) -> None:
+    """Set up the magic spell asynchrously."""
```

Comment:
nit: spelling. Codespell may need tweaks too. No need to look at that for this PR. Just making a note for myself.

Reply:
I have no issue with "nit(picking)", some "collegues" of mine do.  I appreciate the level of exigence.

As far as I know, "magic spell" is the adjective "magic" and the noun "spell" so I do not understand why codespell needs an update for spelling (unless you want to have 'codespell' renamed to 'codespelling' ;-) ) .

Or, as functions should start with a verb it would have to be "spell_tuya_magic" ?.

Reply:
Lol I think asynchronously is missing an n

Reply:
Right, overlooked that one.